### ISSUE TITLE
Warning due to supported source version 'RELEASE_7' from annotation processor

### ISSUE THREAD

**Issue by RohitJ23** [Azure/azure-sdk-for-java#1653]
I get this warning during compilation while using the SDK with JDK 1.8: 
Supported source version 'RELEASE_7' from annotation processor 'com.microsoft.azure.management.apigeneration.LangDefinitionProcessor' less than -source '1.8'.

In the readme, prerequisites include "A Java Developer Kit (JDK), v 1.7 or later".

Is there a reason for having this restriction in place? If not can it be removed?

**Comment by sebeard** [comment]
Java 7 has been EOL since April 2015!! Surely this needs updating to Java 8 ASAP given that the latest release was done in May 2017 (version 1.1.0)

**Comment by jvano** [comment]
If you’re currently running Java apps on-premise or with other JDKs, consider moving it to Zulu on Azure for continued support and maintenance.

For more info:
https://azure.microsoft.com/en-us/blog/microsoft-and-azul-systems-bring-free-java-lts-support-to-azure/

**Comment by jacqueslorentz** [comment]
Hi,

As you recommended, I moved to the Zulu JDK. However I already have the warning, with both JDK 8 and 11.
What can I do, I really appreciate to set the <failOnWarning> flag in the maven compilation options.

Thanks!

**Comment by jvano** [comment]
@selvasingh, can you help us with this case with Zulu? It is in the context of the Azure SDK for Java

**Comment by ChenTanyi** [comment]
@jacqueslorentz do you mean you meet the warning in Zulu JDK 8? Or you meet it in Oracle / Open JDK 8?
And can you share your code snippet about it? I cannot compile any maven project with this error in Oracle JDK 8.

**Comment by jacqueslorentz** [comment]
I use the Zulu JDK 8 and 11, for two distinct projects and I have this warning with both, with this maven plugin configuration in my `pom.xml`:
```xml
...
<build>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-compiler-plugin</artifactId>
                    <version>3.8.1</version>
                    <configuration>
                        <release>${java.version}</release>
                        <showWarnings>true</showWarnings>
                        <!--<failOnWarning>true</failOnWarning> CANNOT COMPILE WITH IT-->
                    </configuration>
                </plugin>
                ...
           </plugins>
      </pluginManagement>
</build>
```
We need to use the Zulu JDK to match with Azure Pipelines environnement.

**Comment by gnikwo** [comment]
Same warning here !
I use Zulu JDK 11 with spring-boot-azure-starter current version.
Any solution ?

**Comment by ChenTanyi** [comment]
The azure-annotations hard code to support RELEASE_7, will fix later (many dependency need to upgrade).
Now, this warning could only be ignore.

**Comment by ChenTanyi** [comment]
fix in 1.29.0

**Comment by martin-mfg** [comment]
@ChenTanyi 
Thanks for your work!
Will this change also come to [`com.microsoft.azure:azure-eventhubs`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/microsoft-azure-eventhubs/pom.xml)?

**Comment by ChenTanyi** [comment]
@martin-mfg This change is included in the [`azure-client-authentication:1.7.0`](https://github.com/Azure/autorest-clientruntime-for-java/releases/tag/v1.7.0) and the [`Fluent Java 1.29.0`](https://github.com/Azure/azure-libraries-for-java/releases/tag/v1.29.0), if you want other specific sdk to be fixed, you could request them to update the client runtime. like [`azure-eventhubs`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/microsoft-azure-eventhubs/pom.xml#L55)

**Comment by yaohaizh** [comment]
@joshfree , @conniey @srnagar @mssfang for eventhub libraries. Thanks.
